src/NHibernate.Test/NHSpecificTest/NH3931/Entities.cs

```diff
@@ -0,0 +1,161 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace NHibernate.Test.NHSpecificTest.NH3931
```

Comment:
> Could you please move tests to NHibernate.Test.Insertordering where the old one lives?

Done, completed with HHH-11585 additional tests. They have added an additional bug on their side by badly simplifying my fix for one-to-one case. So I have not ported that. I have removed the useless dictionary though.

I have not worked on the flawed sorting algorithm today, I will do maybe tomorrow.